ui/app/models/auth-method.js

```diff
@@ -45,8 +45,8 @@ export default class AuthMethodModel extends Model {
   }
 
   get icon() {
-    const authMethods = allMethods().find((backend) => backend.type === this.methodType);
-
+    // methodType refers to the backend type (e.g., "aws", "azure") and is set on a getter.
+    const authMethods = engineDisplayData(this.methodType);
```

Comment:
NIT: Could we rename this const to something more precise, like `engineData`? `authMethods` made sense when we were only returning the auth method, but now that we’re searching across all engines and could return any mount category, a more general name would be clearer.